src/contexts/WebSocketContext.tsx

```
@@ -0,0 +1,188 @@
+  
+  // Don't expose the full logs array to all consumers if it gets too large
+  const lastLogs = logs.length > 100 ? logs.slice(-100) : logs;
+  
+  const value = {
+    logs: lastLogs,
+    error,
+    connected,
+    reconnecting,
+    connect,
+    disconnect
+  };
```

ESLint

[error] The 'value' object (at line 163) passed as the value prop to the Context provider (at line 173) changes every render. To fix this consider wrapping it in a useMemo hook.

(react/jsx-no-constructed-context-values)